modules/apps/frontend-theme/frontend-theme-classic/src/css/custom_properties/_custom_properties_variables.scss

```diff
@@ -47,9 +62,74 @@ $theme-colors-cp: map-deep-merge(
 		success: var(--success),
 		warning: var(--warning),
 	),
-	$theme-colors-cp
+	$custom-properties-theme-colors
 );
 
+// Buttons
+
+@each $key, $map in $custom-properties-button-colors {
+	$normal: ();
+	$hover: ();
+
+	:root {
+		@each $status in (normal, hover) {
+			$prefix: $status + '-';
+
+			@if ($status == normal) {
+				$prefix: '';
+			}
+
+			@each $prop in (background-color, border-color, color) {
+				$alias: $prefix + $prop;
+				$btn-variable: '';
+
+				@if ($prop == background-color) {
+					$alias: $prefix + 'bg';
+				}
+
+				@if (map-has-key($map, #{$alias})) {
+					$btn-variable: map-get($map, #{$alias});
+				} @else if($status == normal) {
+					$btn-variable: map-get($map, #{$prop});
+				} @else {
+					$btn-variable: map-get(map-get($map, #{$status}), #{$prop});
+				}
+
+				@if ($btn-variable != null) {
+					--btn-#{$key}-#{$prefix}#{$prop}: #{$btn-variable};
+
+					@if ($status == normal) {
+						$normal: map-merge(
+							(#{$prop}: var(--btn-#{$key}-#{$prop})),
+							$normal
+						);
+					} @else {
+						$hover: map-merge(
+							(#{$prop}: var(--btn-#{$key}-#{$prefix}#{$prop})),
+							$hover
+						);
+					}
+				}
+			}
+		}
+	}
+
+	$normal: map-merge(
+		(
+			hover: $hover,
+			focus: $hover,
+			active: $hover,
+			disabled: $hover,
+		),
+		$normal
+	);
+
+	$custom-properties-button-maps: map-merge(
+		(#{$key}: $normal),
+		$custom-properties-button-maps
+	);
+}
```

Comment:
I had some hard time trying to fill a `$map` outside the `:root` from the `:root`

**before**
```
$map: ();
:root {
   @each loop.. {
         --print-style

         $map: map-merge((values...), $map);
   }
}
```
`$map` is empty


**current:**
```
$map: ();
@each loop.. {
   :root {
         --print-style
   }

   $map: map-merge((values...), $map);
}
```
`$map` works

@pat270 do you know any trick to make the first example work?

Reply:
Not sure what you are asking, you want to get:
```
	$normal: map-merge(
		(
			hover: $hover,
			focus: $hover,
			active: $hover,
			disabled: $hover,
		),
		$normal
	);

	$custom-properties-button-maps: map-merge(
		(#{$key}: $normal),
		$custom-properties-button-maps
	);
```
to work?

Reply:
I took a look and I think the code above works just fine inside `:root` if that's what you're asking:
```
:root {
	@each $key, $map in $custom-properties-button-colors {
		$normal: ();
		$hover: ();

		@each $status in (normal, hover) {
			$prefix: $status + '-';

			@if ($status == normal) {
				$prefix: '';
			}

			@each $prop in (background-color, border-color, color) {
                                 ...
			}
		}

		$normal: map-merge(
			(
				hover: $hover,
				focus: $hover,
				active: $hover,
				disabled: $hover,
			),
			$normal
		);

		$custom-properties-button-maps: map-merge(
			(#{$key}: $normal),
			$custom-properties-button-maps
		);
	}
}
```

Reply:
I was thinking instead of using nested loops we could do something like:
```
:root {
	@each $key, $value in $custom-properties-button-colors {
		$normal: map-get($custom-properties-button-colors, $key);
		$hover: map-get($normal, hover);

		// normal

		$background-color: setter(
			map-get($normal, bg),
			map-get($normal, background-color)
		);
		$border-color: map-get($normal, border-color);
		$color: map-get($normal, color);

		@if ($background-color) {
			--btn-#{$key}-background-color: #{$background-color};
		}
		@if ($border-color) {
			--btn-#{$key}-border-color: #{$border-color};
		}
		@if ($color) {
			--btn-#{$key}-color: #{$color};
		}

		// hover

		$hover-background-color: setter(
			map-get($normal, hover-bg),
			map-get($hover, background-color)
		);
		$hover-border-color: setter(
			map-get($normal, hover-border-color),
			map-get($hover, border-color)
		);
		$hover-color: setter(
			map-get($normal, hover-color),
			map-get($hover, color)
		);

		@if ($hover-background-color) {
			--btn-#{$key}-hover-background-color: #{$hover-background-color};
		}
		@if ($hover-border-color) {
			--btn-#{$key}-hover-border-color: #{$hover-border-color};
		}
		@if ($hover-color) {
			--btn-#{$key}-hover-color-color: #{$hover-color};
		}
	}
}
```

We could get rid of `$custom-properties-button-maps: ();`.